### PR TITLE
added variable to control creation of challenge dir

### PR DESCRIPTION
**Pull request by muelli**
The test for whether creating the directory has been refactored into a
user controllable variable "pki_create_acme_challenge".

### CODE REVIEW

**Comment by muelli** [comment]
please check whether the test actually makes sense. I've literally cut and pasted it from the tasks file. `ansible_local` looks suspicious to me.

File: defaults/main.yml [hunk 1]
```diff
@@ -233,6 +233,16 @@ pki_acme_challenge_dir: '{{ (ansible_local.nginx.acme_root
...
+
+
+# .. envvar:: pki_create_acme_challenge
+#
+# Whether the directory where the ACME client should store responses
+# should be created.
+pki_create_acme_challenge_dir: '{{
+    (ansible_local|d() and ansible_local.nginx|d()
+     and  ansible_local.nginx.acme|d() and
+          ansible_local.nginx.acme)|bool }}'
```
Comment (drybjed) [inline defaults/main.yml]: I'm not sure what happens if the `ansible_local.nginx.*` dictionary is not set. Perhaps go with something like this instead:
```yaml
pki_create_acme_challenge_dir: '{{ True if (ansible_local|d() and ansible_local.nginx|d() and
                                            ansible_local.nginx.acme|d() and
                                            ansible_local.nginx.acme|bool) else False }}'
```

Comment (ypid) [inline defaults/main.yml]: Maybe you can create the new variable as `pki__create_acme_challenge_dir` to meet the current [DebOps Code Standards Policy](https://docs.debops.org/en/latest/debops-policy/docs/code-standards-policy.html) as well. This means one var less to update for users when the roles gets updated (all semi-automated by anyway).

Comment (muelli) [inline defaults/main.yml]: hah. over at the apache role I intended to make the switch from `d` to `default` and it wasn't worth it ;-)

I'm happy either way.

**Pull request merged**